src/main/java/org/amahi/anywhere/activity/AuthenticationActivity.java

```diff
@@ -284,4 +288,17 @@ public void onBackPressed() {
     protected void attachBaseContext(Context newBase) {
         super.attachBaseContext(LocaleHelper.onAttach(newBase));
     }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        LocationListener.get().unregister(this);
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        asyncTask.cancel(true);
```

Comment:
How is asyncTask variable getting any memory location allocated to it? I can't see a statement like `asyncTask = new AsyncTask()` anywhere.